openwisp-config/files/openwisp.agent

```diff
@@ -370,6 +380,24 @@ get_checksum() {
 		return 2
 	fi
 
+	if is_http_status "$1" 404; then
+		for attempt_no in $(seq 1 "$CHECKSUM_MAX_RETRIES"); do
+			sleep "$CHECKSUM_RETRY_DELAY"
+			$FETCH_COMMAND -i "$CHECKSUM_URL" >"$1"
+			if is_http_status "$1" 404; then
+				if [ "$attempt_no" -eq "$CHECKSUM_MAX_RETRIES" ]; then
```

Comment:
while testing, I noticed that setting the retries to 2 caused the agent to do only 1 retry, I think here we have to use a different test, which check whether the attempt is higher than `CHECKSUM_MAX_RETRIES` otherwise the number of attempts will be `CHECKSUM_MAX_RETRIES - 1`

Reply:
I tested again and both lower bound and upper bound are inclusive in this loop.
Eg:
```bash
CHECKSUM_MAX_RETRIES=2
CHECKSUM_RETRY_DELAY=1
for attempt_no in $(seq 1 "$CHECKSUM_MAX_RETRIES"); do
    sleep $CHECKSUM_RETRY_DELAY
    echo "Fetch count: $attempt_no"
    if [ "$attempt_no" -eq "$CHECKSUM_MAX_RETRIES" ]; then
        echo "Giving up and shutting down (device may have been deleted from controller)."
        exit 0
    fi
done
```
Output:
```
Fetch count: 1
Fetch count: 2
Giving up and shutting down (device may have been deleted from controller).
```

Reply:
I moved the whole checksum logic in the retry loop for consistency with other parts of the code where we have a retry mechanism